### PR TITLE
refactor(shada): remove most goto

### PR DESCRIPTION
**Pull request by dundargoc**
Go To Statement Considered Obnoxious.

### CODE REVIEW

**Comment by dundargoc** [comment]
The title says all which is the end goal, but I still need to figure out 

1. whatever these curly braces even mean in a goto `shada_parse_msgpack_read_next: {}`
2. Why I can't create a function with `ShadaEntry` as a parameter without the compilter warning me it doesn't exist, despite multiple other functions being able to do so.

File: src/nvim/shada.c [hunk 1]
```diff
@@ -1795,31 +1817,31 @@ static ShaDaWriteResult shada_pack_entry(msgpack_packer *const packer, ShadaEntr
...
-        goto shada_pack_entry_error;
+        shada_pack_entry_error(spacker, sbuf);
+        return ret;
       }
     }
   }
   msgpack_packer_free(spacker);
   msgpack_sbuffer_destroy(&sbuf);
   return kSDWriteSuccessfull;
-shada_pack_entry_error:
```
Comment (bfredl) [inline src/nvim/shada.c:1819]: This one is a conventional "goto error" style goto tho, which is commonly used in our codebase.

Comment (dundargoc) [reply src/nvim/shada.c:1819]: ok

**Comment by dundargoc** [comment]
Well, if this is the programming style of neovim and not a solo programmer gone rogue then this PR lost its raison d'etre. I think most gotos are of the style you described. I don't think this PR serves any purpose then. I'll close it if that's OK?

**Comment by dundargoc** [comment]
Go To Statement Considered Useful ;(

**Pull request closed without merging**